src/lib/mina_base/account_update.ml

```diff
@@ -24,7 +24,19 @@ module Authorization_kind = struct
       type t =
             Mina_wire_types.Mina_base.Account_update.Authorization_kind.V1.t =
         | Signature
-        | Proof of (Field.t[@version_asserted])
+        | Proof of
+            (Field.t
+            [@version_asserted]
+            [@to_yojson fun t -> `String (Snark_params.Tick.Field.to_string t)]
+            [@of_yojson
+              function
+              | `String s ->
+                  let field = Snark_params.Tick.Field.of_string s in
```

Comment:
I think this may suffer from the same problem as the current code, but in reverse - if there are any precomputed blocks out there that were produced with `release/3.0.2` or later (with the hex string format) then this code will fail to read them. I think the parser here needs to accept both formats.

The round-trip formatting test for precomputed blocks should fail given this code, in fact, because the test block `hetzner-itn-1-1795.json` has hex authorization kind strings. (If that's okay, then that test will need to be adjusted so the test block has old format authorization kind strings).

Reply:
Good point. And I do confirm the test fails.